SimTKcommon/Simulation/include/SimTKcommon/internal/StateImpl.h

```diff
@@ -636,7 +636,7 @@ class SimTK_SimTKCOMMON_EXPORT StateImpl {
 
     StateImpl& operator=(const StateImpl& src);
 
-    ~StateImpl() {}   // default destructor
+    ~StateImpl() {delete stateLock;}
```

Comment:
The default destructor will be adequate again if you don't heap allocate stateLock.